src/partition/repart.c

```diff
@@ -87,6 +87,9 @@
 /* LUKS2 takes off 16M of the partition size with its metadata by default */
 #define LUKS2_METADATA_SIZE (16*1024*1024)
 
+/* Keep a few MBs free on minimized partitions. */
+#define MINIMIZE_KEEP_FREE (4U*1024U*1024U)
```

Comment:
Why are 4MB of free space needed instead of rounding-up to the next 4KB or so?

A workflow I might want to use is:

1. Create an image with a squashfs file-system, minimize is enabled.
2. Split the partition into a separate file.
3. Serve the file over HTTP for use with systemd-sysupdate.

Since squashfs is already compressed additional transport compression is unnecessary, however 4 MB free space at the end increase the download size by a few percent for smaller partitions.